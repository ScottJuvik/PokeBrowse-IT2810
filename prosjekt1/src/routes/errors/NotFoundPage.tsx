import ditto from '@assets/ditto.png';
import { useNavigate } from 'react-router-dom';
import styles from './NotFoundPage.module.css';

const NotFoundPage = () => {
    const navigate = useNavigate();

    const handleRedirect = () => {
        navigate('/');
    };

    return (
        <div className={styles.errorContainer}>
            <img src={ditto} alt="Ditto" />
            <h1 className="text-4xl">Oh no! It's a Ditto!</h1>
            <p className={`${styles.subtitle} text-xl`}>
                The page you are looking for seems to have transformed and escaped.
            </p>
            <p className={`${styles.text} text-lg`}>
                Maybe it's playing hide and seek, or perhaps you took a wrong turn at Lavender Town.
            </p>

            <button onClick={handleRedirect} className="transition-color text-base">
                Return to Safety
            </button>
        </div>
    );
};

export default NotFoundPage;
