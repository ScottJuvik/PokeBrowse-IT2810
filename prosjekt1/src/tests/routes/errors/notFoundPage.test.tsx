import NotFoundPage from '@/routes/errors/notFoundPage';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

const queryClient = new QueryClient();

describe("notFoundPage", () => {
    it("renders an error 404 page", () => {
        const router = createMemoryRouter(
            [
                {
                    path: '/',
                    element: <NotFoundPage />,
                },
            ],
            {
                initialEntries: ['/'],
            }
        );
        
        const { container } = render(
            <QueryClientProvider client={queryClient}>
                <RouterProvider router={router} />
            </QueryClientProvider>
        );
        
        // assert that image is shown on page
        const image = screen.getByRole('img', { name: "Ditto" });
        expect(image).toBeInTheDocument(); 

        // assert that title text is shown on page
        expect(screen.getByText("Oh no! It's a Ditto!")).toBeInTheDocument();

        // assert that error text is shown on page (two paragraphs)
        // we do this by using a queryselector of the render
        const paragraphs = container.querySelectorAll("p");
        expect(paragraphs).toHaveLength(2);
    });
});