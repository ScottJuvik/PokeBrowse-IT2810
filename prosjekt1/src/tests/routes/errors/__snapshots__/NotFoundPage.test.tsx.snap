// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="_errorContainer_48fbbd"
  >
    <img
      alt="Ditto"
      src="/src/assets/ditto.png"
    />
    <h1
      class="text-4xl"
    >
      Oh no! It's a Ditto!
    </h1>
    <p
      class="_subtitle_48fbbd text-xl"
    >
      The page you are looking for seems to have transformed and escaped.
    </p>
    <p
      class="_text_48fbbd text-lg"
    >
      Maybe it's playing hide and seek, or perhaps you took a wrong turn at Lavender Town.
    </p>
    <button
      class="transition-color text-base"
    >
      Return to Safety
    </button>
  </div>
</DocumentFragment>
`;
